watsononlinestore/watson_online_store.py

```diff
@@ -180,6 +191,93 @@ def setup_conversation_workspace(conversation_client, environ):
         return ret
 
     @staticmethod
+    def setup_discovery_collection(discovery_client,
+                                   environment_id,
+                                   collection_id,
+                                   data_source):
+        """ Ensure that the collection exists in the Watson Discovery service.
+
+        :param discovery_client: discovery service client
+        :param str environment_id: discovery service environment id
+        :param str collection_id: discovery service collection id
+        :param str data_source: name of the discovery data source
+        :return: ID of discovery collection to use
+        :rtype: str
+        :raise Exception: When collection is not found and cannot be created
+        """
+
+        # Data source names
+        AMAZON = "AMAZON"
+        IBM = "IBM_STORE"
+        # Discovery collection names
+        AMAZON_COLLECTION = "amazon-shopping"
+        IBM_COLLECTION = "ibm-logo-store"
+        # Alternate discovery collection names if we must re-create
+        AMAZON_ALT_COLLECTION = "amazon-shopping-alt"
+        IBM_ALT_COLLECTION = "ibm-logo-store-alt"
+        # File path location to discovery html files
+        AMAZON_DATA_PATH = "data/amazon_data_html/"
+        IBM_DATA_PATH = "data/ibm_store_html/"
+
+        # Ensure we a valid environment exists.
+        try:
+            discovery_client.get_environment(environment_id)
+        except Exception:
+            raise Exception("DISCOVERY_ENVIRONMENT_ID=%s is specified in "
+                            "a runtime environment variable, but that "
+                            "environment does not exist." % environment_id)
+
+        # Determine if collection exists.
+        try:
+            collection = discovery_client.get_collection(environment_id,
+                                                         collection_id)
+            if collection:
+                return collection_id
+        except Exception as e:
+            LOG.debug("Watson Discovery - unable to find collection. "
+                      "Error: %s" % repr(e))
```

Comment:
I thought if we got a bad collection ID, we could try and build a new one. So now the only way we build on the fly is if user does not enter a collection id value.